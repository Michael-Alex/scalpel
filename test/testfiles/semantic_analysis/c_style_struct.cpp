struct point
{
	int x;
	int y;
};

struct point
create_point(int x, int y);

struct point
create_point(int x, int y)
{
}

unsigned int
distance(struct point p1, struct point p2);

unsigned int
distance(struct point p1, struct point p2)
{
}

struct point p;

