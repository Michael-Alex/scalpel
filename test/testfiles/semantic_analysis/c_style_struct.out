{
  "id": 0,
  "classes":
  [
    {
      "id": 0,
      "name": "point",
      "destructor":
      {
        "access": "public"
      },
      "variables":
      [
        {
          "id": 0,
          "name": "x",
          "access": "public",
          "type":
          {
            "category": "fundamental type",
            "type": "int"
          }
        },
        {
          "id": 1,
          "name": "y",
          "access": "public",
          "type":
          {
            "category": "fundamental type",
            "type": "int"
          }
        }
      ]
    }
  ],
  "simple functions":
  [
    {
      "id": 0,
      "name": "create_point",
      "return type":
      {
        "category": "class",
        "id": 0
      },
      "parameters":
      [
        {
          "name": "x",
          "type":
          {
            "category": "fundamental type",
            "type": "int"
          }
        },
        {
          "name": "y",
          "type":
          {
            "category": "fundamental type",
            "type": "int"
          }
        }
      ]
    },
    {
      "id": 1,
      "name": "distance",
      "return type":
      {
        "category": "fundamental type",
        "type": "unsigned int"
      },
      "parameters":
      [
        {
          "name": "p1",
          "type":
          {
            "category": "class",
            "id": 0
          }
        },
        {
          "name": "p2",
          "type":
          {
            "category": "class",
            "id": 0
          }
        }
      ]
    }
  ],
  "variables":
  [
    {
      "id": 0,
      "name": "p",
      "type":
      {
        "category": "class",
        "id": 0
      }
    }
  ]
}
